metrics/metricinfo.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/fdb-exporter/db"
+	"io"
+	"time"
+
+	"github.com/tigrisdata/fdb-exporter/models"
+	ulog "github.com/tigrisdata/fdb-exporter/util/log"
+	"github.com/uber-go/tally"
+	promreporter "github.com/uber-go/tally/prometheus"
+)
+
+type MetricInfo struct {
+	groups []Collectable
+	status models.FullStatus
+	Closer io.Closer
+	scopes map[string]tally.Scope
+}
+
+var Reporter promreporter.Reporter
```

Comment:
This might be more personal choice. But I would make the reporter part of the MetricInfo struct. And then in main I would do this:

```
   http.Serve(metricInfo.Reporter)
```

Pass it as a reference to it.